Update election fencing doc. (#3097)
tarantool/doc: doc/book/replication/repl_leader_elect.rst

This can lead to the split situation if the other nodes elect a new leader upon losing the connectivity to the previous one.

The issue is resolved in Tarantool version :doc:`2.10.0 </release/2.10.0>` by introducing the leader *fencing* mode.
The mode can be switched by the :ref:`election_fencing_mode <repl_leader_elect_config>` configuration parameter.

> When the fencing is set to ``soft`` or ``strict``, the leader resigns its leadership if it has less than the

Reviewer: ```suggestion
When the fencing is set to ``soft`` or ``strict``, the leader resigns its leadership if it has less than
```
`replication_synchro_quorum` is a name, so no article is needed.

Author: Fixed.

:ref:`replication_synchro_quorum <repl_leader_elect_config>` of alive connections to the cluster nodes.
The resigning leader receives the status of a follower in the current election term and becomes read-only.

> Leader *fencing* can be turned off by setting :ref:`election_fencing_mode <repl_leader_elect_config>` configuration parameter to ``off``.

Reviewer: ```suggestion
Leader *fencing* can be turned off by setting the :ref:`election_fencing_mode <repl_leader_elect_config>` configuration parameter to ``off``.
```
And here the article "the" is attached to the word "parameter", so it is required.

> In ``soft`` mode connection is considered dead if the where no responses for
> :ref:`4*replication_timeout <cfg_replication-replication_timeout>` both on current leader and followers.
> 
> In ``strict`` mode connection is considered dead if the where no responses
> for :ref:`2*replication_timeout <cfg_replication-replication_timeout>` on current leader and
> :ref:`4*replication_timeout <cfg_replication-replication_timeout>` on followers.

Reviewer: ```suggestion
In ``soft`` mode, a connection is considered dead if there are no responses for
:ref:`4*replication_timeout <cfg_replication-replication_timeout>` seconds both on the current leader and the followers.

In ``strict`` mode, a connection is considered dead if there are no responses
for :ref:`2*replication_timeout <cfg_replication-replication_timeout>` seconds on the current leader and for
:ref:`4*replication_timeout <cfg_replication-replication_timeout>` seconds on the followers.
```

  meaning that each node becomes a leader immediately after voting for itself.
  It is the best to set up this option value to the ``(<cluster size> / 2) + 1``.
  Otherwise, there is no guarantee that there is only one leader at a time.

> * ``election_fencing_mode`` -- specifies the:ref:`leader fencing mode <repl_leader_elect_fencing>`.

Reviewer: ```suggestion
* ``election_fencing_mode`` -- specifies the :ref:`leader fencing mode <repl_leader_elect_fencing>`.
```
Missing space
